.github/workflows/OntologyValidator.yml

```diff
@@ -0,0 +1,77 @@
+# This workflow adds guids to any newly added entity types in the ontology and validates the ontology
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Ontology Validator
+
+on:
+  push:
+    branches: [ master ]
+    paths: 
+      - ontology/yaml/**
+
+  pull_request:
+    branches: [ master ]
+    paths: 
+      - ontology/yaml/**
+
+jobs:
+  build:
+  
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+
+    steps:
+    - name: Checkout the branch the PR is coming from
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ github.event.pull_request.head.repo.full_name }}
+        ref: ${{ github.event.pull_request.head.ref }}
+    
+    - name: Checkout head
+      if: ${{ github.event_name == 'push' }}
+      uses: actions/checkout@v3
+    
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
```

Comment:
I recommend using another version as 3.7 end of support is in 3 months
https://www.python.org/downloads

see "Active Python Releases"